### PR TITLE
#2091 filter out split service from weighted cluster in route config

### PR DESCRIPTION
**Pull request by addozhang**
**Description**:

described in #2091 

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [X]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

**Comment by addozhang** [comment]
reopen

**Comment by shashankram** [comment]
@michelleN this PR is attempting to solve a problem your changes are also attempting to address. Could you share your PRs for reference just in case this PR is not required because of the routes refactor.

**Comment by draychev** [comment]
@addozhang Thank you so much for working on this!!  This is a good change!

@shashankram @michelleN I propose we a) merge this PR b) write an integration (higher level than unittest) test to confirm the Envoys receive the expected configuration  and so then when @michelleN finishes the work on her side - the test will confirm there's no regression.

**Comment by shashankram** [comment]
> @addozhang Thank you so much for working on this!! This is a good change!
> 
> @shashankram @michelleN I propose we a) merge this PR b) write an integration (higher level than unittest) test to confirm the Envoys receive the expected configuration and so then when @michelleN finishes the work on her side - the test will confirm there's no regression.

I am not opposed to merging this change, but at the very least I'd suggest to add unit tests testing the particular scenario being fixed, in this PR. 

I'll let @michelleN and @snehachhabria comment on the correctness of the fix.

**Comment by draychev** [comment]
@addozhang would you be so kind to check the **DCO** Details link in the GitHub checks: 
image

You can read more about how to signoff here:  https://github.com/openservicemesh/osm/pull/2092/checks?check_run_id=1422971556

We also have a bit about that in [our CONTRIBUTING doc](https://github.com/openservicemesh/osm/blob/main/CONTRIBUTING.md#sign-your-work)

**Comment by addozhang** [comment]
@draychev done

File: pkg/envoy/rds/response.go [hunk 1]
```diff
@@ -37,6 +37,16 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_disco
...
 
+	allTrafficSplits, _, _, _, _ := catalog.ListSMIPolicies()
+	isTrafficSplitService := func(svc service.MeshService) bool {
+		for _, split := range allTrafficSplits {
+			if split.Namespace == svc.Namespace && split.Spec.Service == svc.Name {
+				return true
+			}
+		}
+		return false
+	}
```
Comment (snehachhabria) [inline pkg/envoy/rds/response.go:48]: Could we carve this out as a separate function so that it is unit testable ? 

Once separated please add a unit test

**Review by snehachhabria — COMMENTED** [review]
@addozhang Thanks a lot for the fix, the change looks good. I had one minor comment, once that's addressed we can check in this code.

File: pkg/envoy/rds/response.go (hunk 1 above)
Comment (addozhang) [reply pkg/envoy/rds/response.go:48]: Yes, will provide ut laster.

**Comment by draychev** [comment]
Hi @addozhang 
Now that #2110 is merged - does it make sense to close this PR here?

**Comment by addozhang** [comment]
> Hi @addozhang 
> 
> Now that #2110 is merged - does it make sense to close this PR here?

yes. plz

**Comment by shashankram** [comment]
Closing as per https://github.com/openservicemesh/osm/pull/2092#issuecomment-738465408

**Pull request closed without merging**